scalafix/rules/src/main/scala/utils/GlobalProxyService.scala

```diff
@@ -1,14 +1,13 @@
-package scala.meta.internal.proxy
+package scala.tools.nsc.interactive
 
 import scala.reflect.internal.util.Position
-import scala.tools.nsc.interactive.Global
-
-object GlobalProxyService {
-  def typedTreeAt(g: Global, pos: Position): g.Tree = {
-    // NOTE(olafur) clearing `unitOfFile` fixes a bug where `typedTreeAt` would
-    // produce symbols with erroneous `.info` signatures.
-    g.unitOfFile.clear()
-    g.typedTreeAt(pos)
-  }
+import scala.reflect.internal.util.SourceFile
+
+class GlobalProxyService[G <: Global](g: G, sourceFile: SourceFile) {
+  private val _ = g.reloadSource(sourceFile)
+  private val _ = g.typedTree(sourceFile, true)
```

Comment:
maybe add a comment to explain why it is important to do this only once for each sourceFile

Reply:
ok